source/common/http/http2/codec_impl.cc

```diff
@@ -1324,15 +1363,15 @@ bool ServerConnectionImpl::checkInboundFrameLimits() {
                    "received in this HTTP/2 session.",
                    connection_);
     stats_.inbound_empty_frames_flood_.inc();
-    return false;
+    return bufferFloodError("Too many consecutive frames with an empty payload");
   }
 
   if (inbound_priority_frames_ > max_inbound_priority_frames_per_stream_ * (1 + inbound_streams_)) {
     ENVOY_CONN_LOG(trace,
                    "error reading frame: Too many PRIORITY frames received in this HTTP/2 session.",
                    connection_);
     stats_.inbound_priority_frames_flood_.inc();
-    return false;
+    return bufferFloodError("Too many PRIORITY frames");
```

Comment:
Should there be new tests to expect on this error text, or is this already covered?

Reply:
The old tests did not check for the exception message as it was replaced with something generic after returning through the nghttp2 stack. Now we can get the original message and I have modified tests to check for it.